### PR TITLE
test(client): test JS PlanetScale connector

### PR DESCRIPTION
**Pull request by Jolg42**
WIP

Based on https://github.com/prisma/prisma/pull/20507

Related
https://github.com/prisma/team-orm/issues/275
https://github.com/prisma/team-orm/issues/258

### CODE REVIEW

File: packages/client/tests/functional/17810-direct-url-missing/prisma/_schema.ts [hunk 1]
```diff
@@ -1,20 +1,19 @@
+import { computeSchemaHeader } from '../../_utils/computeSchemaHeader'
 import { idForProvider } from '../../_utils/idForProvider'
 import testMatrix from '../_matrix'
 
-export default testMatrix.setupSchema(({ provider }) => {
+export default testMatrix.setupSchema(({ provider, providerFlavor }): string => {
+  const schemaHeader = computeSchemaHeader({
+    provider,
+    providerFlavor,
+    directUrl: `env("DIRECT_DATABASE_URI_${providerFlavor ? providerFlavor : provider}")`,
```
Comment (millsp) [inline packages/client/tests/functional/17810-direct-url-missing/prisma/_schema.ts:9]: Maybe we can make this a boolean instead?

**Comment by millsp** [comment]
Would it make sense to extract the refactored helpers and `_schema.ts` into a separate PR for easier review?

**Comment by Jolg42** [comment]
@millsp 🤔 it could be tricky and I need to focus on more testing, so I won't try anything on that side

File: packages/client/tests/functional/17810-direct-url-missing/prisma/_schema.ts (hunk 1 above)
Comment (Jolg42) [reply packages/client/tests/functional/17810-direct-url-missing/prisma/_schema.ts:9]: It's only used in one place at the moment. Maybe can be done later.

**Comment by Jolg42** [comment]
The story continues here https://github.com/prisma/prisma/pull/20991

**Pull request closed without merging**